alphapulldown/analysis_pipeline/get_good_inter_pae.py

```diff
@@ -108,50 +108,67 @@ def main(argv):
+                    result_subdir = os.path.join(FLAGS.output_dir, job)
+                    
+                    with open(ranking_debug_path, 'r') as f:
+                        ranking_data = json.load(f)
+                    try:
+                        best_model = ranking_data['order'][0]
+                        data = ranking_data
+
+                        if "iptm" in data and "iptm+ptm" in data:
+                            iptm_ptm_score = data['iptm+ptm'][best_model]
+                            
+                            try:
+                                pae_mtx, iptm_score = obtain_pae_and_iptm(result_subdir, best_model)
+                                seq_lengths = obtain_seq_lengths(result_subdir)
+                                check = examine_inter_pae(pae_mtx, seq_lengths, cutoff=FLAGS.cutoff)
+                                mpDockq_score, plddt_per_chain = obtain_mpdockq(os.path.join(FLAGS.output_dir, job))
+                                
+                                if check:
+                                    good_jobs.append(job)
+                                    score_df = pdb_analyser(os.path.join(pi_score_output_path, job), pae_mtx, plddt_per_chain)
+                                    score_df['jobs'] = job
+                                    score_df['iptm_ptm'] = iptm_ptm_score
+                                    score_df['iptm'] = iptm_score
+                                    score_df['pDockQ/mpDockQ'] = mpDockq_score
+                                    
+                                    for i in ['pDockQ/mpDockQ', 'iptm', 'iptm_ptm', 'jobs']:
+                                        score_df.insert(0, i, score_df.pop(i))
+                                        
+                                    output_df = pd.concat([score_df, output_df])
+                            except Exception as e:
+                                logging.error(f"Error processing PAE and iPTM for job {job}: {e}")
+                    except Exception as e:
+                        logging.error(f"Error getting the best model name from ranking_debug.json for job :{job} : {e}")
+                except Exception as e:
+                    logging.error(f"Error processing ranking_debug.json for job {job}: {e}")
+            else:
+                logging.warning(f"{job} does not have ranking_debug.json. Skipped.")
+        except Exception as e:
+            logging.error(f"Error processing job {job}: {e}")
+        finally:
+            logging.info(f"done for {job} {count} out of {len(jobs)} finished.")
+
     if len(good_jobs) == 0:
         logging.info(
             f"Unfortunately, none of your protein models had at least one PAE on the interface below your cutoff value : {FLAGS.cutoff}.\n Please consider using a larger cutoff.")
     else:
-        unwanted_columns = ['pi_score','pdb',' pvalue']
+        unwanted_columns = ['pdb',' pvalue', 'pvalue']
```

Comment:
Do you have 'pvalue' in unwanted columns defined twice intentionally?

Reply:
> Do you have 'pvalue' in unwanted columns defined twice intentionally?

Yea it's quite annoying that I found in the PI score output files, some files write column name 'pvalue' but some write ' pvalue' with an empty space before letter p. I added twice just to be 100% sure I remove the pvalue column, regardless of the formats. It's a bit ugly solution I agree